src/client/consumer.ts

```diff
@@ -70,29 +74,35 @@ class Consumer implements vscode.Disposable {
      * Received messages and/or errors are emitted via events.
      */
     async start(): Promise<void> {
+        const partition = this.options.partition;
+        const fromOffset = this.options.fromOffset;
+        const topic = this.options.topicId;
         this.kafkaClient = createKafka(this.options);
-        this.consumer = this.kafkaClient.consumer({ groupId: `vscode-kafka-${this.options.clusterId}-${this.options.topic}`, retry: { retries: 3 }});
+        this.consumer = this.kafkaClient.consumer({ groupId: this.options.consumerGroupId, retry: { retries: 3 } });
         await this.consumer.connect();
-        await this.consumer.subscribe({ topic: this.options.topic, fromBeginning: this.options.fromOffset ===  "earliest" })
 
-        await this.consumer.run({
-            eachMessage: async ({ topic , partition, message }) => {
+        const subscribeOptions = this.createSubscribeOptions(topic, fromOffset);
+        await this.consumer.subscribe(subscribeOptions)
+
+        this.consumer.run({
```

Comment:
It seems you need to define a custom [partitionAssigner](https://kafka.js.org/docs/consuming#custom-partition-assigner), if the user defined the partition, else the consumer will be assigned to all partitions:

```
{"level":"INFO","timestamp":"2021-02-01T11:11:32.397Z","logger":"kafkajs","message":"[ConsumerGroup] Consumer has joined the group","groupId":"foo-earliest","memberId":"vscode-kafka-9b8fc2c0-bd42-4519-babd-6817f4053d7c","leaderId":"vscode-kafka-9b8fc2c0-bd42-4519-babd-6817f4053d7c","isLeader":true,"memberAssignment":{"6pr8ehzk-test":[0,1,2,3,4]},"groupProtocol":"RoundRobinAssigner","duration":3479}
```

Reply:
My understanding is to filter from partitions, without partitionAssigner I have the same behaviour than conduktor. I disable this partition assigned for the moment, please give me feedback with or without partitionAssigner. To enablepartitionAssigner please remove comment /* partitions */ in line https://github.com/jlandersen/vscode-kafka/pull/98/files#diff-f66af15c392b2023b98da968390a2eecdd539cbc17ac8ca4b23eb8db3e6de95fR78

Reply:
Ok now I'm using partitionsAssigner.